staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go

```diff
@@ -104,6 +104,9 @@ type Request interface {
 	// was idle when the value was calculated, but might no longer be
 	// accurate by the time the client examines that value.
 	Finish(execute func()) (idle bool)
+
+	// OnDispatch is called when the request is to be dispatched
+	OnDispatch()
```

Comment:
It is a bit odd to make clients responsible for this bit of accounting, which is otherwise private to the queueset implementation.

Reply:
Refined this part.